src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/GraphNodes/DependenciesGraphProvider.cs

```diff
@@ -152,7 +145,7 @@ Graph IGraphProvider.Schema
         {
             try
             {
-                await EnsureInitializedAsync().ConfigureAwait(false);
+                await InitializeAsync().ConfigureAwait(true);   // InitializeAsync switches to the UI thread, we want to get off it
```

Comment:
The base class already prevents double initialization here?

Reply:
Yep, **OnceInitialized**OnceDisposedAsync - this line will start initialization or join it.

This code was also problematic as it let you proceed while the class was still initializing. We were previously only hooking an event - so no big deal, but now we store a field that must be set before we proceed.